libsolidity/interface/StandardCompiler.cpp

```diff
@@ -171,16 +171,22 @@ bool hashMatchesContent(string const& _hash, string const& _content)
 bool isArtifactRequested(Json::Value const& _outputSelection, string const& _artifact, bool _wildcardMatchesExperimental)
 {
 	static set<string> experimental{"ir", "irOptimized", "wast", "ewasm", "ewasm.wast"};
-	for (auto const& artifact: _outputSelection)
-		/// @TODO support sub-matching, e.g "evm" matches "evm.assembly"
-		if (artifact == _artifact)
+	for (auto const& artifactJson: _outputSelection)
+	{
+		string const& artifact = artifactJson.asString();
+		if (_artifact == artifact || (
+			_artifact.size() > artifact.size() &&
+			_artifact.substr(0, artifact.size()) == artifact &&
+			_artifact.at(artifact.size()) == '.'
+		))
```

Comment:
This part would be much nicer to read with more distinctive names :)
```suggestion
	for (auto const& selectedArtifactJson: _outputSelection)
	{
		string const& selectedArtifact = selectedArtifactJson.asString();
		if (_artifact == selectedArtifact || (
			_artifact.size() > selectedArtifact.size() &&
			_artifact.substr(0, selectedArtifact.size()) == selectedArtifact &&
			_artifact.at(selectedArtifact.size()) == '.'
		))
```